src/transformers/models/gpt2/configuration_gpt2.py

```diff
@@ -249,25 +249,25 @@ def generate_dummy_inputs(
 
                 batch, seqlen = common_inputs["input_ids"].shape
                 # Not using the same length for past_key_values
-                past_key_values_length = seqlen + 2
+                past_key_values_length = 1
                 past_shape = (
+                    2,
                     batch,
                     self.num_attention_heads,
                     past_key_values_length,
                     self._config.hidden_size // self.num_attention_heads,
                 )
                 ordered_inputs["past_key_values"] = [
-                    (torch.zeros(past_shape), torch.zeros(past_shape)) for _ in range(self.num_layers)
+                    torch.zeros(past_shape, dtype=torch.float32) for _ in range(self.num_layers)
```

Comment:
Does it work with latest `transformers` versions?
Because the expecte input is a tuple of tuples of tensors. It might work because indexing would behave the same way?

Reply:
Indeed, it would be nice to see an explicit example with a GPT-2 checkpoint, with `past_key_values` passed to the model